pkg/operator/controllers/checkers/internetchecker/controller_test.go

```diff
@@ -0,0 +1,131 @@
+package internetchecker
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+import (
+	"context"
+	"errors"
+	"reflect"
+	"testing"
+	"time"
+
+	operatorv1 "github.com/openshift/api/operator/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+
+	"github.com/Azure/ARO-RP/pkg/operator"
+	arov1alpha1 "github.com/Azure/ARO-RP/pkg/operator/apis/aro.openshift.io/v1alpha1"
+	arofake "github.com/Azure/ARO-RP/pkg/operator/clientset/versioned/fake"
+	checkercommon "github.com/Azure/ARO-RP/pkg/operator/controllers/checkers/common"
+	"github.com/Azure/ARO-RP/pkg/util/cmp"
+	utillog "github.com/Azure/ARO-RP/pkg/util/log"
+)
+
+type fakeChecker func(URLs []string) error
+
+func (fc fakeChecker) Check(URLs []string) error {
+	return fc(URLs)
+}
+
+func TestReconcile(t *testing.T) {
+	ctx := context.Background()
+	urlsToCheck := []string{"https://fake-url-for-test-only.xyz"}
+
+	tests := []struct {
+		name               string
+		controllerDisabled bool
+		checkerReturnErr   error
+		wantCondition      operatorv1.ConditionStatus
+		wantErr            string
+		wantResult         reconcile.Result
+	}{
+		{
+			name:          "no errors",
+			wantCondition: operatorv1.ConditionTrue,
+			wantResult:    reconcile.Result{RequeueAfter: time.Hour},
+		},
+		{
+			name:             "error making a request",
+			checkerReturnErr: errors.New("fake error from checker"),
+			wantCondition:    operatorv1.ConditionFalse,
+			wantErr:          "fake error from checker",
+			wantResult:       reconcile.Result{RequeueAfter: time.Hour},
+		},
+		{
+			name:               "controller disabled",
+			controllerDisabled: true,
+			wantResult:         reconcile.Result{},
+		},
+	}
+
+	roleToConditionTypeMap := map[string]string{
+		"master": arov1alpha1.InternetReachableFromMaster,
+		"worker": arov1alpha1.InternetReachableFromWorker,
+	}
+	for _, testRole := range []string{operator.RoleMaster, operator.RoleWorker} {
```

Comment:
since our `conditionType()` switches over 'three values' should we add a nonworker/nonmaster role in this slice? - might be overkill though.